core/vm/noop.go

```diff
@@ -23,48 +23,105 @@ import (
-func (NoopStateDB) SetState(common.Address, common.Hash, common.Hash)                  {}
-func (NoopStateDB) Suicide(common.Address) bool                                        { return false }
-func (NoopStateDB) HasSuicided(common.Address) bool                                    { return false }
-func (NoopStateDB) Exist(common.Address) bool                                          { return false }
-func (NoopStateDB) Empty(common.Address) bool                                          { return false }
-func (NoopStateDB) RevertToSnapshot(int)                                               {}
-func (NoopStateDB) Snapshot() int                                                      { return 0 }
-func (NoopStateDB) AddLog(*types.Log)                                                  {}
-func (NoopStateDB) AddPreimage(common.Hash, []byte)                                    {}
+// CreateAccount dummy method
+func (NoopStateDB) CreateAccount(common.Address) {}
+
+// SubBalance dummy method
+func (NoopStateDB) SubBalance(common.Address, *big.Int) {}
+
+// AddBalance dummy method
+func (NoopStateDB) AddBalance(common.Address, *big.Int) {}
+
+// GetBalance dummy method
+func (NoopStateDB) GetBalance(common.Address) *big.Int { return nil }
+
+// GetNonce dummy method
+func (NoopStateDB) GetNonce(common.Address) uint64 { return 0 }
+
+// SetNonce dummy method
+func (NoopStateDB) SetNonce(common.Address, uint64) {}
+
+// GetCodeHash dummy method
+func (NoopStateDB) GetCodeHash(common.Address) common.Hash { return common.Hash{} }
+
+// GetCode dummy method
+func (NoopStateDB) GetCode(common.Address) []byte { return nil }
+
+// SetCode dummy method
+func (NoopStateDB) SetCode(common.Address, []byte) {}
+
+// GetCodeSize dummy method
+func (NoopStateDB) GetCodeSize(common.Address) int { return 0 }
+
+// AddRefund dummy method
+func (NoopStateDB) AddRefund(uint64) {}
+
+// GetRefund dummy method
+func (NoopStateDB) GetRefund() uint64 { return 0 }
+
+// GetState dummy method
+func (NoopStateDB) GetState(common.Address, common.Hash) common.Hash { return common.Hash{} }
+
+// SetState dummy method
+func (NoopStateDB) SetState(common.Address, common.Hash, common.Hash) {}
+
+// Suicide dummy method
+func (NoopStateDB) Suicide(common.Address) bool { return false }
+
+// HasSuicided dummy method
+func (NoopStateDB) HasSuicided(common.Address) bool { return false }
+
+// Exist dummy method
+func (NoopStateDB) Exist(common.Address) bool { return false }
+
+// Empty dummy method
+func (NoopStateDB) Empty(common.Address) bool { return false }
+
+// RevertToSnapshot dummy method
+func (NoopStateDB) RevertToSnapshot(int) {}
+
+// Snapshot dummy method
+func (NoopStateDB) Snapshot() int { return 0 }
+
+// AddLog dummy method
+func (NoopStateDB) AddLog(*types.Log) {}
+
+// AddPreimage dummy method
+func (NoopStateDB) AddPreimage(common.Hash, []byte) {}
+
+// ForEachStorage dummy method
```

Comment:
Please revert these changes. These don't add any meaningful information to the code. On the other hand, the many lines bloat up the code and make it harder to see that this is a noop mock construct.